opendatahub-io/odh-dashboard
# [RHOAIENG-4660] Compare runs - Manage runs page

Closes: [RHOAIENG-4660](https://issues.redhat.com/browse/RHOAIENG-4660)

## Description
Manage runs page uses selected runs from the Compare runs page initially as pre-selected table rows (runs), and the user has the ability to select up to 10 total (and change what was initially selected if they want). Selections are maintained between pages, cancelling uses what was initially selected by the Compare runs page, and "Update" will update the selections and bring the user back to the Compare runs page.

## Tests
Cypress tests for main interactions

## How to test
1. Navigate to `http://localhost:4010/experiments/jps-fun-world/85c9d729-bf63-441f-af87-d6898e5db5f0/compareRuns/add`
2. Verify the initial Experiment filter is set to the experiment associated with the ID in your URL path
3. Select up to 10 table rows on a single or multiple pages
4. "Update" and verify you're navigated to the Compare runs page with URL search string updated with all run valid IDs
5. Navigate to the same path from step 1
6. "Cancel" and verify you're navigated to the Compare runs page with no "runs" in the search param
7. Navigate to the same path from step 1, except with `?runs=` following with a comma delimited list of run IDs to verify initial rows are checked (assuming those IDs are valid)
8. Verify filter, row links, pagination, sorting works for the Manage runs table
9. Verify selections between pages persist
10. Verify breadcrumbs bring you to the correct locations

<img width="1487" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/96431149/d3c5f2df-1876-454b-8c94-791bcb7ed305">
<img width="1156" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/96431149/1f2ac84f-8fac-450c-b408-c60bcd7f5644">

(cc @yannnz)

## Request review criteria:
Self checklist (all need to be checked):

If you have UI changes: 

After the PR is posted & before it merges:
